perpVects/QFPHelpers.cpp

```diff
@@ -37,19 +37,51 @@ std::ostream& operator<<(std::ostream& os, const unsigned __int128 &i){
   return os;
 }
 
-std::string
-getSortName(sort_t val){
-  switch(val){
-  case lt:
-    return "lt";
-  case gt:
-    return "gt";
-  case bi:
-    return "bi";
-  case def:
-    return "us";
-  default:
-    return "something bad happened, undefined sort type";
+namespace FPHelpers{
+    auto
+  swap_float_int(unsigned int val) -> float{
```

Comment:
On Wednesday, August 24, 2016, Geof Sawaya <geof.sawaya@gmail.com> wrote:

> Nope.  The auto is used to make it a polymorphic type, obfuscating the
> return type.  Otherwise the compiler complains about different return types


I know that sounds weird, I'll double check (it was 4am)

>
> On Wednesday, August 24, 2016, Michael Bentley <notifications@github.com
> <javascript:_e(%7B%7D,'cvml','notifications@github.com');>> wrote:
>
>> In perpVects/QFPHelpers.cpp
>> <https://github.com/Geof23/QFP/pull/13#discussion_r76153200>:
>>
>> > -std::string
>> > -getSortName(sort_t val){
>> > -  switch(val){
>> > -  case lt:
>> > -    return "lt";
>> > -  case gt:
>> > -    return "gt";
>> > -  case bi:
>> > -    return "bi";
>> > -  case def:
>> > -    return "us";
>> > -  default:
>> > -    return "something bad happened, undefined sort type";
>> > +namespace FPHelpers{
>> > +    auto
>> > +  swap_float_int(unsigned int val) -> float{
>>
>> Why the use of auto when you know the return type? The only reason for
>> that kind of notation is to be able to use the parameters in a decltype()
>> call to determine the return type.
>>
>> I suggest just using for example:
>>
>> float swap_float_int(uint32_t val) { ... }
>>
>> —
>> You are receiving this because you authored the thread.
>> Reply to this email directly, view it on GitHub
>> <https://github.com/Geof23/QFP/pull/13/files/57db3da0327a58ab91d6a9ee1c70501608c4db5b#r76153200>,
>> or mute the thread
>> <https://github.com/notifications/unsubscribe-auth/AFiXD7gTBg7fPOZWVIuhQQxgnJVkq99Lks5qjMjjgaJpZM4Jr1FF>
>> .
>>
>